src/effects/backends/builtin/compressoreffect.cpp

```diff
@@ -152,75 +194,62 @@ void CompressorEffect::processChannel(
         const EffectEnableState enableState,
         const GroupFeatureState& groupFeatures) {
     Q_UNUSED(groupFeatures);
-    Q_UNUSED(enableState);
+
+    if (enableState == EffectEnableState::Enabling) {
+        pState->clear(engineParameters);
+    } else {
+        pState->calculateCoeffsIfChanged(engineParameters, m_pAttack->value(), m_pRelease->value());
+    }
 
     SINT numSamples = engineParameters.samplesPerBuffer();
 
     // Compression
-    applyCompression(pState, engineParameters, pInput, pOutput);
+    CSAMPLE* pGainBuffer = applyCompression(pState, engineParameters, pInput, pOutput);
 
     // Auto make up
     if (m_pAutoMakeUp->toInt() == static_cast<int>(AutoMakeUp::AutoMakeUpOn)) {
-        applyAutoMakeUp(pState, pInput, pOutput, numSamples);
+        applyAutoMakeUp(pState, numSamples, pGainBuffer);
     }
 
     // Output gain
     CSAMPLE gain = static_cast<CSAMPLE>(db2ratio(m_pLevel->value()));
-    SampleUtil::applyGain(pOutput, gain, numSamples);
+
+    // Finally applying all gains
+    for (SINT i = 0; i < engineParameters.samplesPerBuffer(); i++) {
+        pOutput[i] = pInput[i] * pGainBuffer[i] * gain;
+    }
 }
 
-void CompressorEffect::applyAutoMakeUp(CompressorGroupState* pState,
-        const CSAMPLE* pInput,
-        CSAMPLE* pOutput,
-        const SINT& numSamples) {
-    CSAMPLE rmsInput = SampleUtil::rms(pInput, numSamples);
-    if (rmsInput > CSAMPLE_ZERO) {
-        CSAMPLE_GAIN makeUpGainState = pState->previousMakeUpGain;
+void CompressorEffect::applyAutoMakeUp(
+        CompressorGroupState* pState, SINT numSamples, CSAMPLE* pGainBuffer) {
+    CSAMPLE rmsGain = SampleUtil::rms(pGainBuffer, numSamples);
 
-        CSAMPLE rmsOutput = SampleUtil::rms(pOutput, numSamples);
-        CSAMPLE_GAIN makeUp = rmsInput / rmsOutput;
+    CSAMPLE_GAIN makeUpGainState = pState->makeUpGainState;
 
-        // smoothing
-        makeUpGainState = kMakeUpAttackCoeff * makeUp + (1 - kMakeUpAttackCoeff) * makeUpGainState;
+    // smoothing
+    makeUpGainState = pState->makeUpCoeff * rmsGain + (1 - pState->makeUpCoeff) * makeUpGainState;
 
-        pState->previousMakeUpGain = makeUpGainState;
-        SampleUtil::applyGain(pOutput, makeUpGainState, numSamples);
-    }
+    pState->makeUpGainState = makeUpGainState;
+    SampleUtil::applyGain(pGainBuffer, 1 / makeUpGainState, numSamples);
```

Comment:
I think we need to ramp here as well:
```suggestion
    SampleUtil::applyRampingGain(pGainBuffer, 1 / pState->makeUpGainState, 1/makeUpGainState, numSamples);
    pState->makeUpGainState = makeUpGainState;
```

Reply:
Fixed!